### ISSUE TITLE
1.17.1: Visible <chat=X> in messages

### ISSUE THREAD

**Issue by TheIntelloBox** [kangarko/ChatControl#1456]
BungeeCord: true
MySQL: true

I configured my channels and DiscordSRV.
However, when someone sends a message, there is a suffix on Discord after his message. For example: "kangarko: hello<chat=4455445-dsssqd78-sddsqds-45456>"

Moreover, my messages don't have any format

Best regards,
TheIntelloBox

**Comment by ElBananaa** [comment]
> Hello,
> 
> I configured my channels and DiscordSRV. However, when someone sends a message, there is a suffix on Discord after his message. For example: "kangarko: hello<chat=4455445-dsssqd78-sddsqds-45456>"
> 
> Moreover, my messages don't have any format
> 
> Best regards, TheIntelloBox

Hey!
Please provide us your `/chc debug` file.
Also make sure the plugin is up-to-date, and the required dependencies (In this case ProtocolLib and vault) are also installed and up-to-date.
Finally, please follow the bug-report format next time.

**Comment by TheIntelloBox** [comment]
Hello,

Which website can I use to send you the ZIP ?

**Comment by ElBananaa** [comment]
Hey,
You should be able to drag and drop the file here, if you can't, I believe we give you the name of a website in the chat when you generate the debug file. Anyways, if we don't, just upload the file on wetransfer and send us the link

**Comment by TheIntelloBox** [comment]
Here it is: https://ufile.io/57ccg2vj

**Comment by ElBananaa** [comment]
> Hello,
> 
> I configured my channels and DiscordSRV. However, when someone sends a message, there is a suffix on Discord after his message. For example: "kangarko: hello<chat=4455445-dsssqd78-sddsqds-45456>"
> 
> Moreover, my messages don't have any format
> 
> Best regards, TheIntelloBox

Thanks!
When does it happen exactly? It always happens, or only if on a specific player/on a specific server/on a specific channel?
Could you please provide us a screenshot of one (or more) message(s) concerned by this issue?

Also, please update to v10.7.9.

**Comment by TheIntelloBox** [comment]
Hello,

It happens for every player and for every message. By the way, DiscordSRV doesn't allow bungeecord chat, is it possible with ChatControl Red to have messages from every server in the same channel ? (by using the global chat channel)

![ChatControl 2](https://user-images.githubusercontent.com/16572147/141445058-63e439e7-56ec-4e76-b813-8ba89a752a04.jpg)

**Comment by TheIntelloBox** [comment]
Okay it seems like it isn't only a Discord bug.
![Bug](https://user-images.githubusercontent.com/16572147/141559645-57ee7365-150d-41df-8299-aa58049e3a32.jpg)

**Comment by ElBananaa** [comment]
> Okay it seems like it isn't only a Discord bug. ![Bug](https://user-images.githubusercontent.com/16572147/141559645-57ee7365-150d-41df-8299-aa58049e3a32.jpg)

Hmmm...
Could you please provide us your latest.log?

**Comment by TheIntelloBox** [comment]
I have one error: org.mineacademy.chatcontrol.lib.exception.FoException: Report: Malformed configuration! Key 'Messages.Apply_On' in settings.yml must be List but got String: 'timed'

**Comment by ElBananaa** [comment]
> I have one error: org.mineacademy.chatcontrol.lib.exception.FoException: Report: Malformed configuration! Key 'Messages.Apply_On' in settings.yml must be List but got String: 'timed'

Oh, set : `Apply_On: [timed]` instead, it'll fix the error. As for the discord thing, please wait for Kangarko

**Comment by kangarko** [comment]
Which chat format is it that causes this? When does it happen? I cannot find it in your settings.

**Comment by TheIntelloBox** [comment]
It is the global channel

**Comment by TheIntelloBox** [comment]
And it happens for every message. ;)

**Comment by kangarko** [comment]
I am sorry for the delay, unfortunately I am unable to reproduce this:

![a](https://i.imgur.com/m9jKm9N.png)

Please use 10.8.0. Can you try disabling ProtocolLib or ViaVersion for one testing round to see if the problem does not go away?

**Comment by kangarko** [comment]
Closing this to cleanup, please comment below if you need anything and we'll help you!

**Comment by TheIntelloBox** [comment]
The problem comes from ProtocolLib. How to fix my issue ?

**Comment by ElBananaa** [comment]
> The problem comes from ProtocolLib. How to fix my issue ?

Hey,
Update ProtocolLib to either the latest version available on spigot (if 1.17 or under), or use the latest dev build available (if 1.18 or above)

**Comment by TheIntelloBox** [comment]
> > The problem comes from ProtocolLib. How to fix my issue ?
> 
> Hey, Update ProtocolLib to either the latest version available on spigot (if 1.17 or under), or use the latest dev build available (if 1.18 or above)

I already did that. ;)

**Comment by ElBananaa** [comment]
> > > The problem comes from ProtocolLib. How to fix my issue ?
> > 
> > 
> > Hey, Update ProtocolLib to either the latest version available on spigot (if 1.17 or under), or use the latest dev build available (if 1.18 or above)
> 
> I already did that. ;)

Please provide us your up-to-date `/chc debug` file. I'm unable to reproduce your issue

**Comment by TheIntelloBox** [comment]
> > > > The problem comes from ProtocolLib. How to fix my issue ?
> > > 
> > > 
> > > Hey, Update ProtocolLib to either the latest version available on spigot (if 1.17 or under), or use the latest dev build available (if 1.18 or above)
> > 
> > 
> > I already did that. ;)
> 
> Please provide us your up-to-date `/chc debug` file. I'm unable to reproduce your issue

https://ufile.io/l7pj656m

**Comment by ElBananaa** [comment]
> > > > > The problem comes from ProtocolLib. How to fix my issue ?
> > > > 
> > > > 
> > > > Hey, Update ProtocolLib to either the latest version available on spigot (if 1.17 or under), or use the latest dev build available (if 1.18 or above)
> > > 
> > > 
> > > I already did that. ;)
> > 
> > 
> > Please provide us your up-to-date `/chc debug` file. I'm unable to reproduce your issue
> 
> https://ufile.io/l7pj656m

Thanks,
Please update chatcontrol to the latest version available on mcmarket, same thing with bungeecontrol.
Once this is done, you have two choices: Either disable viaversion, or disable our protocolLib integration.
To disabled ProtocolLib in chatcontrol, go to the settings.yml file, and set `ProtocolLib.Listen_For_Packets` & `Book_Anti_Crash` to false.

Keep in mind that disabling this will also cause the message deleting feature, packet rules book filtering & tab completion to break on chatcontrol's side.

**Comment by TheIntelloBox** [comment]
> feature, packet rules book filtering & tab completion to break on chatcontrol's side.

Thanks I'll do that. There will be no fix for that issue ?

**Comment by ElBananaa** [comment]
> > feature, packet rules book filtering & tab completion to break on chatcontrol's side.
> 
> Thanks I'll do that. There will be no fix for that issue ?

Well, I don't know of this is something that could be fixed. Plugins like ViaVersion/Geyser are often pretty tricky when it comes to packet handling, which is why a lot of packet-based plugins don't work properly when these plugins are installed, such as anti cheats, as example.

@kangarko will certainly give you more infos about that ^^

**Comment by kangarko** [comment]
That's a separate issue. Open a new ticket, include clear description of the issue since I am not sure if I understand, plus how can I reproduce and a /chc debug output. Thank you!

**Comment by TheIntelloBox** [comment]
Actually, that's the same issue... 

When someone sends a message in Minecraft, the Discord message will have a suffix "<chat=number>"
For example, if I send in Minecraft hello, it will be on Discord: "hello<chat=4455445-dsssqd78-sddsqds-45456>"

After disabling ProtocolLib the issue disappears.

**Comment by kangarko** [comment]
Looks like an issue with our [X] feature, can you remove the [X] from chat.yml to see if it goes away?

**Comment by TheIntelloBox** [comment]
delete-chat:
    Message: '&8[&4X&8] &f'
    Receiver_Permission: chatcontrol.removemessages
    Hover:
    - '&cRemove this message from chat for all players.'
    Run_Command: /{label} internal remove {message_uuid}

Here ?

**Comment by ElBananaa** [comment]
> delete-chat: Message: '&8[&4X&8] &f' Receiver_Permission: chatcontrol.removemessages Hover: - '&cRemove this message from chat for all players.' Run_Command: /{label} internal remove {message_uuid}
> 
> Here ?

Yes, this part

**Comment by TheIntelloBox** [comment]
Do I put ### like:

#delete-chat:#
#Message: '&8[&4X&8] &f'#
#Receiver_Permission: chatcontrol.removemessages#
#Hover:#
#- '&cRemove this message from chat for all players.'#
#Run_Command: /{label} internal remove {message_uuid}#

Or like that: 
Message: ''
Receiver_Permission: chatcontrol.removemessages
Hover:
- '&cRemove this message from chat for all players.'
Run_Command: /{label} internal remove {message_uuid}

**Comment by ElBananaa** [comment]
> Do I put ### like:
> 
> #delete-chat:# #Message: '&8[&4X&8] &f'# #Receiver_Permission: chatcontrol.removemessages# #Hover:# #- '&cRemove this message from chat for all players.'# #Run_Command: /{label} internal remove {message_uuid}#
> 
> Or like that: Message: '' Receiver_Permission: chatcontrol.removemessages Hover:
> 
> * '&cRemove this message from chat for all players.'
>   Run_Command: /{label} internal remove {message_uuid}

You can either fully remove this part for the file, or comment it using # in front of each line ^^

**Comment by TheIntelloBox** [comment]
I still have the problem. ;)

**Comment by TheIntelloBox** [comment]
Someone wants to come on my server and Discord server to see the bug by yourself ?

**Comment by ElBananaa** [comment]
Hey!
Do you use Geyser (To allow bedrock clients), and/or ViaVersion? If so, please try to disable these plugins to see if the issue persists

Also, please provide us an updated version of your debug file.

**Comment by TheIntelloBox** [comment]
I don't use Geyser and disabling ViaVersion or ViaBackwards or ViaRewind doesn't solve the problem.

https://ufile.io/2cg76ca4

**Comment by ElBananaa** [comment]
> Reference in n

Could you try to reproduce the issue with paper instead of pufferfish?

**Comment by TheIntelloBox** [comment]
> > Reference in n
> 
> Could you try to reproduce the issue with paper instead of pufferfish?

In which file ?

**Comment by ElBananaa** [comment]
> > > Reference in n
> > 
> > 
> > Could you try to reproduce the issue with paper instead of pufferfish?
> 
> In which file ?
I'm talking about your server software. You are using Pufferfish instead of PaperMC. Try to start your server with paper to see ut that changes anything

**Comment by TheIntelloBox** [comment]
I'm using PaperMC (dev build 1.18.1-117) not Pufferfish.

**Comment by ElBananaa** [comment]
Alright, I'm reopening this issue so Kangarko can have a look at it.

**Comment by kangarko** [comment]
You're on an outdated version, please use 10.9.2 or wait for the next release coming up tonight.

I cannot find the code for "chat=" being added anywhere in the sources of the current version.
